### ISSUE TITLE
U55N support?

### ISSUE THREAD

**Issue by OhGodAPet** [Xilinx/open-nic-shell#21]
I would love to see support for the C1100 (U55N) in this!

**Comment by dloghin** [comment]
\+ 1 here

I tried to add support for au55n but I encountered some issues. Please see below.

Looking at the project structure, we need to add the following for au55n:
- add a folder under _board_files/Xilinx_ (e.g., name it au55n and place the board files there). I copied the board files from /opt/xilinx/platforms/xilinx_u55n_gen3x4_xdma_1_202110_1/hw/hw.xsa (I decompressed the archive). The /opt/xilinx/platforms/xilinx_u55n_gen3x4_xdma_1_202110_1 folder is created by installing the deb packages from https://www.xilinx.com/products/accelerators/varium/c1100.html. Please see https://github.com/dloghin/open-nic-shell/tree/main/board_files/Xilinx/au55n
- add a folder under _constr_. I copied the files from au50 and adapted the pins based on the info in _part0_pins.xml_ under _board_files_. Please see https://github.com/dloghin/open-nic-shell/tree/main/constr/au55n
- add a new file named _cmac_usplus_0_au55n.tcl_ under _src/cmac_subsystem/vivado_ip_. I adapted this from au50. Please see https://github.com/dloghin/open-nic-shell/blob/main/src/cmac_subsystem/vivado_ip/cmac_usplus_0_au55n.tcl.

This last file is the source of my build error:
[ERROR] ... Value '4x25' is out of the range for parameter 'Line Rate(NUM_LANES)' for IP 'cmac_usplus_0' . Valid values are - 10x10
I also get these warnings:
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'CMAC_CAUI4_MODE' from '0' to '1' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'INCLUDE_RS_FEC' from '0' to '1' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'CMAC_CORE_SELECT' from 'CMACE4_X0Y3' to 'CMACE4_X0Y4' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'GT_GROUP_SELECT' from 'X0Y0~X0Y9' to 'X0Y28~X0Y31' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE9_GT_LOC' from 'X0Y8' to 'NA' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE8_GT_LOC' from 'X0Y7' to 'NA' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE7_GT_LOC' from 'X0Y6' to 'NA' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE6_GT_LOC' from 'X0Y5' to 'NA' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE5_GT_LOC' from 'X0Y4' to 'NA' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE4_GT_LOC' from 'X0Y3' to 'X0Y31' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE3_GT_LOC' from 'X0Y2' to 'X0Y30' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE2_GT_LOC' from 'X0Y1' to 'X0Y29' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE1_GT_LOC' from 'X0Y0' to 'X0Y28' has been ignored for IP 'cmac_usplus_0'
WARNING: [IP_Flow 19-3374] An attempt to modify the value of disabled parameter 'LANE10_GT_LOC' from 'X0Y9' to 'NA' has been ignored for IP 'cmac_usplus_0'

Looking at the other boards, au55n is missing the preset.xml file (this should be under board_files/Xilinx/au55n/). Where to get this file from? I could not find it under /opt/xilinx/platforms/xilinx_u55n_gen3x4_xdma_1_202110_1/.

In summary, it would be great if the AMD-Xilinx team can add support for au55n (Varium C1100). Thank you.

**Comment by cneely-amd** [comment]
Hi, pull request #24 merged in today adds support for U55N.
